src/agent/onefuzz-agent/src/tasks/generic/input_poller.rs

```diff
@@ -27,6 +27,26 @@ pub enum State<M> {
     Processed(M),
 }
 
+impl<M: PartialEq> PartialEq for State<M> {
+    fn eq(&self, other: &State<M>) -> bool {
+        use State::*;
+
+        match (self, other) {
+            (Ready, Ready) => true,
+            (Polled(l), Polled(r)) =>
+                l == r,
+            (Parsed(l0, l1), Parsed(r0, r1)) =>
+                l0 == r0 && l1 == r1,
+            (Downloaded(l0, l1, l2, l3), Downloaded(r0, r1, r2, r3)) =>
+                l0 == r0 && l1 == r1 && l2 == r2 && l3.path() == r3.path(),
```

Comment:
The last clause is why we can't just derive `PartialEq`: we ust compare the two `tempfile::TempDir` values via their projection to `path()`, since `TempDir: !PartialEq`.

In a future refactor we can wrap `TempDir` in a newtype that impls `PartialEq`, or upstream the same.